spec/unit/action/deploy/environment_spec.rb

```diff
@@ -261,9 +266,13 @@
         let(:purge_levels) { [:deployment] }
 
         it "only logs about purging deployment" do
+          expect(subject).to receive(:visit_environment).and_wrap_original do |original, env, &block|
+            expect(env.logger).to_not receive(:debug).with(/purging unmanaged puppetfile content/i)
```

Comment:
So you know what's bizarre? I don't think this expectation is firing either. I flipped it to `to receive` and the test still passes.

Reply:
However, if I drop the `expect` on `env.logger`, the `expect(true)` does trip...